### DISCUSSION TITLE
How do I use $$ to click on the links of ul.li...li

### DISCUSSION THREAD

**Discussion by fatkobra** [webdriverio/webdriverio#6544] [Q&A] [answered]
Hi, I am learning WebdriverIO.
I am reading through the documentation: https://webdriver.io/docs/api/element/$$

I am trying to test the following code:
```html
<details class="site-navigation__menu" open="">
        <summary role="button" tabindex="0" aria-expanded="true">
            <span>Menu</span>
        </summary>
        <ul>
           
    <li class="site-navigation__menu-item  " data-menu-id="694862">
            <a href="https://www.example.com/organisation/" title="Organisation">Organisation</a>
            </li>

    <li class="site-navigation__menu-item  " data-menu-id="694866">
            <a href="https://www.example.com/investors/" title="Investors">Investors</a>
            </li>

    <li class="site-navigation__menu-item  " data-menu-id="694867">
            <a href="https://www.example.com/products/" title="Products">Products</a>
            </li>

    <li class="site-navigation__menu-item  " data-menu-id="974084">
            <a href="https://www.example.com/insights/" title="Insights">Insights</a>
            </li>

    <li class="site-navigation__menu-item  " data-menu-id="694869">
            <a href="https://www.example.com/responsibility/" title="Responsibility">Responsibility</a>
            </li>

    <li class="site-navigation__menu-item  " data-menu-id="97019">
            <a href="https://www.example.com/careers/" title="Careers">Careers</a>
            </li>

    <li class="site-navigation__menu-item  " data-menu-id="187840">
            <a href="https://www.example.com/media/" title="Media">Media</a>
            </li>

    <li class="site-navigation__menu-item  " data-menu-id="97020">
            <a href="https://www.visma.com/log-in/" title="Log in">Log in</a>
            </li>

        </ul>
    </details>
```

Initially I start with the getText:

```js
describe('Testing webpage', () => {
        it('Click on the Investors link', () => {
        browser.url('/')
        const text = $('.site-navigation__menu');
        console.log(text.$$('li')[2].$('a').getText());
    });
});
```

But my focus is on how can I click on the links.
Any tips?

**Comment by christian-bromann** [comment]
How about:

```js
text.$$('li')[2].$('a').click()
```

**Reply by fatkobra** [reply]
Hi Christian,
Thanks for a very useful tip.
But I get the following error.
`[0-0] 2021-03-04T09:14:00.676Z INFO webdriver: Retrying 3/3
2021-03-04T09:14:00.676Z INFO webdriver: [POST] http://localhost:9515/session/a1ba480c6e32d1eb9a9595dea5beafca/element/bc916c4c-c22b-4d50-a13a-b77c2bdcd335/click
[0-0] 2021-03-04T09:14:00.698Z ERROR webdriver: **Request failed with status 400 due to element not interactable: element not interactable**
  (Session info: chrome=88.0.4324.150)
[0-0] element not interactable in "Visma.com page should get text a menu link"
element not interactable
  (Session info: chrome=88.0.4324.150)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at Context.<anonymous> (/home/brreg.no/extsds/Documents/WEBDRIVERio/webdriverio-test/test/specs/visma.js:56:33)
[0-0] 2021-03-04T09:14:00.704Z INFO webdriver: COMMAND deleteSession()
[0-0] 2021-03-04T09:14:00.705Z INFO webdriver: [DELETE] http://localhost:9515/session/a1ba480c6e32d1eb9a9595dea5beafca
`
Could it be something wrong with my installation?

**Reply by christian-bromann** [reply]
no, the element is not interactable which means a different element is laying on top of it or something else that doesn't allow to interact with it

**Reply by fatkobra** [reply]
Any ideas how I can work around this?

**Reply by fatkobra** [reply]
I succeded in testing the same elements using cypress. I do not get an error with cypress.
Could this be a bug with WebdriverIO?

**Reply by christian-bromann** [reply]
@fatkobra can you provide a reproducible example?

**Reply by fatkobra** [reply]
Hei Christian,
I have tried on these 2 webpages:

https://www.blazemeter.com/
and try clicking on Product | Customers | Resources

https://www.visma.com/
click on Organisation (it works), but the rest do not work: Investors | Products | Insights | Responsibility | Careers | Media

For both pages I can click on all the elements testing using cypress, but cannot with webdriverio (clicking on Organisation works).

**Comment by christian-bromann** [comment, marked as answer]
Running:
```js
describe('My Login application', () => {
    it('should login with valid credentials', () => {
        browser.url('https://www.blazemeter.com/');
        
        $('=PRODUCT').click();
        console.log(browser.getUrl())
    });
});
```
works just fine for me.

**Reply by fatkobra** [reply]
Hey man,
thank you so much for your help.
I initially re-used your code and it did not work.
You are the expert and you it worked, so I thought somethinh wrong with my environment. So I started with:

`browser.setWindowSize(width, height)`

And it worked. Here is the code:

`describe('My Login application', () => {
    it('Click on product', () => {
        browser.setWindowSize(1280,720)
        browser.url('https://www.blazemeter.com/');
        
        $('=PRODUCT').click();
        console.log(browser.getUrl())
        browser.pause(3000)
    });

    it('Click on Customers', () => {
        browser.setWindowSize(1280,720)
        browser.url('https://www.blazemeter.com/');
        
        $('=CUSTOMERS').click();
        console.log(browser.getUrl())
        browser.pause(3000)
    });
});
`
